src/Raven.Server/Documents/Handlers/PullReplicationHandler.cs

```diff
@@ -32,5 +42,62 @@ public async Task DefinePullReplication()
                     json[nameof(OngoingTask.TaskId)] = index;
                 }, statusCode: HttpStatusCode.Created);
         }
+
+        [RavenAction("/databases/*/admin/certificates/feature", "PUT", AuthorizationStatus.Operator)]
+        public Task PutFeatureCertificateOnEdge()
+        {
+            return PutCertificate(PullReplicationDefinition.GetPrefix(Database.Name), includePrivateKey: true);
+        }
+
+        [RavenAction("/databases/*/info/remote-task/tcp", "GET", AuthorizationStatus.RestrictedAccess)]
+        public Task GetRemoteTaskTcp()
+        {
+            var remoteTask = GetStringQueryString("remote-task");
+            var database = GetStringQueryString("database");
+            Authenticate(HttpContext, ServerStore, database, remoteTask);
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var output = Server.ServerStore.GetTcpInfoAndCertificates(HttpContext.Request.GetClientRequestedNodeUrl());
+                context.Write(writer, output);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public static void Authenticate(HttpContext httpContext, ServerStore serverStore, string database, string remoteTask)
+        {
+            // TODO: think of some use cases like:
+            // 1. HTTP, but we defined cert.
```

Comment:
Should allow, this is already handled externally and not here.
If we don't have HTTPS, we always allow.